pkg/util/dockerutils.go

```diff
@@ -175,3 +178,39 @@ func NetExists(client *docker.Client, name string) bool {
 	}
 	return false
 }
+
+// ContainerWait provides a wait loop to check for container in "healthy" status.
+func ContainerWait(timeout time.Duration, labels map[string]string) error {
+	timedOut := time.After(timeout * time.Second)
+	tick := time.Tick(500 * time.Millisecond)
+	for {
+		select {
+		case <-timedOut:
+			return errors.New("health check timed out")
+		case <-tick:
+			container, err := FindContainerByLabels(labels)
+			if err != nil {
+				return errors.New("failed to query container")
+			}
+			status := GetContainerHealth(container)
+			if status == "healthy" {
+				return nil
+			}
+		}
+	}
+}
+
+// GetContainerHealth retrieves the status of a given container. The status string returned
+// by docker contains uptime and the health status in parenths. This function will filter the uptime and
+// return only the health status.
+func GetContainerHealth(container docker.APIContainers) string {
+	status := container.Status
+	re := regexp.MustCompile(`\(([^\)]+)\)`)
+	match := re.FindString(status)
+	match = strings.Trim(match, "()")
+	pre := "health: "
+	if strings.HasPrefix(match, pre) {
+		match = strings.TrimPrefix(match, pre)
+	}
+	return match
```

Comment:
it feels a bit strange to me that there's no potential for error in any of this.